nemo/collections/asr/data/huggingface/hf_audio_to_text.py

```diff
@@ -0,0 +1,592 @@
+        eos_id: int | None = None,
+        pad_id: int = 0,
+        return_sample_id: bool = False,
+        id_key: str | None = None,
+        channel_selector: ChannelSelectorType | None = None,
+        normalize_db: float | None = None,
+        ref_channel: int | None = None,
+        global_rank: int = 0,
+        world_size: int = 0,
+        shuffle_n: int = 0,
+        shuffle_seed: int = 1234,
+        parser: Union[str, Callable] = 'en',
+        blank_index: int = -1,
+        unk_index: int = -1,
+        normalize: bool = True,
+        normalize_text: bool = False,
+        symbols_to_keep: Optional[str] = None,
+    ) -> None:
+        self.labels = labels
+
+        parser = parsers.make_parser(
+            labels=labels, name=parser, unk_id=unk_index, blank_id=blank_index, do_normalize=normalize
+        )
+
+        super().__init__(
+            audio_key=audio_key,
+            text_key=text_key,
+            sample_rate_key=sample_rate_key,
+            hf_data_cfg=hf_data_cfg,
+            parser=parser,
+            sample_rate=sample_rate,
+            augmentor=augmentor,
+            trim=trim,
+            bos_id=bos_id,
+            eos_id=eos_id,
+            pad_id=pad_id,
+            return_sample_id=return_sample_id,
+            id_key=id_key,
+            channel_selector=channel_selector,
+            normalize_db=normalize_db,
+            ref_channel=ref_channel,
+            global_rank=global_rank,
+            world_size=world_size,
+            shuffle_n=shuffle_n,
+            shuffle_seed=shuffle_seed,
+            normalize_text=normalize_text,
+            symbols_to_keep=symbols_to_keep,
+        )
+
+
+class HFIterableAudioToBPEDataset(_HFIterableAudioTextDataset):
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+               """
+        return {
+            'audio_signal': NeuralType(('B', 'T'), AudioSignal()),
+            'a_sig_length': NeuralType(tuple('B'), LengthsType()),
+            'transcripts': NeuralType(('B', 'T'), LabelsType()),
+            'transcript_length': NeuralType(tuple('B'), LengthsType()),
+            'sample_id': NeuralType(tuple('B'), LengthsType(), optional=True),
+        }
+
+    def __init__(
+        self,
+        audio_key: str,
+        text_key: str,
+        sample_rate_key: str,
+        hf_data_cfg: DictConfig,
+        tokenizer: 'nemo.collections.common.tokenizers.TokenizerSpec',
+        sample_rate: int,
+        augmentor: 'nemo.collections.asr.parts.perturb.AudioAugmentor' = None,
+        trim: bool = False,
+        return_sample_id: bool = False,
+        id_key: str | None = None,
+        channel_selector: ChannelSelectorType | None = None,
+        normalize_db: float | None = None,
+        ref_channel: int | None = None,
+        global_rank: int = 0,
+        world_size: int = 0,
+        shuffle_n: int = 0,
+        shuffle_seed: int = 1234,
```

Comment:
Do we need datasets by default ? I don't remember

Reply:
could you please elaborate the question?